fvm/evm/evm_test.go

```diff
@@ -49,66 +40,72 @@ func TestEVMRun(t *testing.T) {
                               return EVM.run(tx: tx, coinbase: coinbase)
                           }
                         `,
-						service.HexWithPrefix(),
-					))
-
-					gasLimit := uint64(100_000)
-
-					txBytes := testAccount.PrepareSignAndEncodeTx(t,
-						testContract.DeployedAt.ToCommon(),
-						testContract.MakeStoreCallData(t, big.NewInt(num)),
-						big.NewInt(0),
-						gasLimit,
-						big.NewInt(1),
-					)
-
-					tx := cadence.NewArray(
-						ConvertToCadence(txBytes),
-					).WithType(stdlib.EVMTransactionBytesCadenceType)
-
-					coinbase := cadence.NewArray(
-						ConvertToCadence(testAccount.Address().Bytes()),
-					).WithType(stdlib.EVMAddressBytesCadenceType)
-
-					accountCodes := map[common.Location][]byte{}
-					var events []cadence.Event
-
-					runtimeInterface := &TestRuntimeInterface{
-						Storage:           backend,
-						OnResolveLocation: SingleIdentifierLocationResolver(t),
-						OnUpdateAccountContractCode: func(location common.AddressLocation, code []byte) error {
-							accountCodes[location] = code
-							return nil
-						},
-						OnGetAccountContractCode: func(location common.AddressLocation) (code []byte, err error) {
-							code = accountCodes[location]
-							return code, nil
-						},
-						OnEmitEvent: func(event cadence.Event) error {
-							events = append(events, event)
-							return nil
-						},
-						OnDecodeArgument: func(b []byte, t cadence.Type) (cadence.Value, error) {
-							return json.Decode(nil, b)
-						},
-					}
-
-					result, err := inter.ExecuteScript(
-						runtime.Script{
-							Source:    script,
-							Arguments: EncodeArgs([]cadence.Value{tx, coinbase}),
-						},
-						runtime.Context{
-							Interface:   runtimeInterface,
-							Environment: interEnv,
-							Location:    common.ScriptLocation{},
-						},
-					)
-					require.NoError(t, err)
-
-					assert.Equal(t, cadence.Bool(true), result)
+							chain.ServiceAddress().HexWithPrefix(),
+						))
+
+						gasLimit := uint64(100_000)
+
+						txBytes := testAccount.PrepareSignAndEncodeTx(t,
+							testContract.DeployedAt.ToCommon(),
+							testContract.MakeStoreCallData(t, big.NewInt(num)),
+							big.NewInt(0),
+							gasLimit,
+							big.NewInt(0),
+						)
+
+						tx := cadence.NewArray(
+							ConvertToCadence(txBytes),
+						).WithType(stdlib.EVMTransactionBytesCadenceType)
+
+						coinbase := cadence.NewArray(
+							ConvertToCadence(testAccount.Address().Bytes()),
+						).WithType(stdlib.EVMAddressBytesCadenceType)
+
+						script := fvm.Script(code).WithArguments(
+							json.MustEncode(tx),
+							json.MustEncode(coinbase),
+						)
+
+						executionSnapshot, output, err := vm.Run(
+							ctx,
+							script,
+							snapshot)
+						require.NoError(t, err)
+						require.NoError(t, output.Err)
+						assert.Equal(t, cadence.Bool(true), output.Value)
+
+						_ = executionSnapshot
+						// snapshot = snapshot.Append(executionSnapshot)
```

Comment:
Do these lines still need to be turned into assertions about the execution state changes?